File: cpp/1493.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int i=0;
        int count = 0;
        int max_count = 0;
        int d = 0;
        int j=0;
        while(j<nums.size()) {
            if(nums[j] == 1) 
                j++;
            else if(d == 0) {
                d++;
                j++;
            }
            else {
                if(nums[i] == 0)
                    d--;
                i++;
            }
            count = max(count, j-i-1);
            
        }
        return count;
    }
};

File: python/1493.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        i=0
        j=0
        count=0
        d=0
        while(j<len(nums)):
            if(nums[j]==1):
                j+=1
            elif(d==0):
                d+=1
                j+=1
            else:
                while d>0:
                    if(nums[i]==0):
                        d-=1
                    i+=1
            count = max(count, j-i-1)
        return count